lib/active_merchant/billing/gateways/blue_snap.rb

```diff
@@ -214,7 +214,7 @@ def add_credit_card(doc, card)
       end
 
       def add_metadata(doc, options)
-        transaction_meta_data = options.fetch(:transaction_meta_data, {})
+        transaction_meta_data = options[:transaction_meta_data] || []
```

Comment:
Nice catch n' fix! Did you check that your new test fails as expected if this code change here isn't in place?

Reply:
Yes - apologies for not putting that in the description.

```
Error: test_successful_purchase_with_metadata_nil(BlueSnapTest): NoMethodError: undefined method `empty?' for nil:NilClass
```